mobilecoind/strategies/burn.py

```diff
@@ -224,13 +224,22 @@ def wait_for_monitor_to_sync(self, monitor_id, poll_interval=0.1):
     submit_response = client.submit_tx(tx_proposal)
     print(submit_response)
 
-    time.sleep(3)
 
-    client.wait_for_monitor_to_sync(monitor)
-    new_balance = client.get_balance(monitor, args.token_id)
+    success = False
+    for i in range(20):
+        logging.info(f"Waiting for burn transaction, attempt #{i}")
+        time.sleep(1)
+
+        client.wait_for_monitor_to_sync(monitor)
+        new_balance = client.get_balance(monitor, args.token_id)
+
+        if new_balance == balance - args.value - args.fee:
+            logging.info("Burn transaction was successful")
+            success = True
+            break
+        else:
+            logging.error(f"Burn transaction appears to be unsuccessful: new_balance={new_balance} balance={balance} args.value={args.value} args.fee={args.fee}")
 
-    if new_balance == balance - args.value - args.fee:
-        logging.info("Burn transaction was successful")
-    else:
-        logging.error("Burn transaction appears to be unsuccessful")
+    if not success:
+        logging.error("Giving up :(")
```

Comment:
:wrench: Python for loops have an else statement that is exercised when they are not broken out of [official docs](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) another [wording](https://book.pythontips.com/en/latest/for_-_else.html#else-clause)
```suggestion
    for i in range(20):
        logging.info(f"Waiting for burn transaction, attempt #{i}")
        time.sleep(1)

        client.wait_for_monitor_to_sync(monitor)
        new_balance = client.get_balance(monitor, args.token_id)

        if new_balance == balance - args.value - args.fee:
            logging.info("Burn transaction was successful")
            break
        else:
            logging.error(f"Burn transaction appears to be unsuccessful: new_balance={new_balance} balance={balance} args.value={args.value} args.fee={args.fee}")

    else:
        logging.error("Giving up :(")
```

Reply:
Nice, I forgot about that. Thanks!

Reply:
fwiw, in some projects they recommend not to use this, because a lot of developers don't know this and don't understand what it means when they read it

Reply:
idk, don't want to block progress.

we can try to resolve questions about using for ... else in another github issue perhaps

I know that this was forbidden in the tesla autopilot project when i worked there. I don't believe any of their code is open source.